### ISSUE TITLE
Rule: `no-generic-names-in-reducers`

### ISSUE THREAD

**Issue by sergeysova** [effector/eslint-plugin#72]
I propose to add check to abandon names in reducer arguments like `state`, `payload`, `data`, `result` and so on.
Arguments named with context should allow to effectively understand code.

Decline:

```ts
const $list = createStore([])
  .on(added, (state, payload) => [...state, payload])
```

Allowed:

```ts
const $list = createStore([])
  .on(added, (list, item) => [...list, item])
```

**Comment by igorkamyshev** [comment]
Personally, I think this rule isn't connected with Effector. Yeah, we should choose names of variables wisely. In any code. But why should we put in here?

E.g. anyone can configure rule https://eslint.org/docs/rules/id-denylist for the project.

**Comment by igorkamyshev** [comment]
@sergeysova what do you think?

**Comment by ainursharaev** [comment]
@igorkamyshev We use the `id-denylist` rule to cover such cases in a project I work on, and have no problems with it

```
"id-denylist": ["warn", "data", "payload"]
```
